### PR TITLE
kitty: update to 0.27.1.

### PR DESCRIPTION
**Pull request by sirkhancision**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, x86_64-glibc

### CODE REVIEW

File: srcpkgs/kitty/template [hunk 1]
```diff
@@ -1,33 +1,38 @@
...
+version=0.27.1
+revision=1
 pycompile_dirs="usr/lib/kitty"
-hostmakedepends="pkg-config python3 wayland-devel wayland-protocols"
-makedepends="gettext-devel glfw-devel harfbuzz-devel libxkbcommon-devel
- python3-devel wayland-devel wayland-protocols librsync-devel libcanberra-devel openssl-devel"
-depends="kitty-terminfo-${version}_${revision}"
+hostmakedepends="go pkg-config python3 wayland-devel wayland-protocols"
+makedepends="dbus-devel fontconfig-devel freetype-devel gettext-devel glfw-devel harfbuzz-devel lcms2-devel libXcursor-devel libXi-devel libXinerama-devel libXrandr-devel libcanberra-devel libpng-devel librsync-devel librsync-devel libxcb-devel libxkbcommon-devel openssl-devel ncurses-devel python3-devel python3-setuptools wayland-devel wayland-protocols zlib-devel"
+depends="kitty-terminfo-${version}_${revision} kitty-shell-integration-${version}_${revision} python3-Pygments"
```
Comment (classabbyamp) [inline srcpkgs/kitty/template:8]:
```suggestion
depends="kitty-terminfo>=${version}_${revision} kitty-shell-integration>=${version}_${revision} python3-Pygments"
```
should depend on kitty-kitten too

File: srcpkgs/kitty/template [hunk 2]
```diff
@@ -36,7 +41,22 @@ do_install() {
 
 kitty-terminfo_package() {
 	short_desc+=" - terminfo data"
+	depends="ncurses"
```
Comment (classabbyamp) [inline srcpkgs/kitty/template:44]: why is this needed? none of the other terminfo packages need it

File: srcpkgs/kitty/template [hunk 3]
```diff
@@ -36,7 +41,22 @@ do_install() {
...
+
+kitty-shell-integration_package() {
+	short_desc+=" - shell integration scripts"
+	pkg_install() {
+		vmove usr/lib/kitty/shell-integration
+	}
+}
+
+kitty-kitten_package() {
+	short_desc+=" - standalone executable"
```
Comment (classabbyamp) [inline srcpkgs/kitty/template:58]:
```suggestion
	short_desc+=" - kitten client"
```
[alpine](https://pkgs.alpinelinux.org/package/edge/community/aarch64/kitty-kitten) describes it as this

File: srcpkgs/kitty/template (hunk 2 above)
Comment (sirkhancision) [reply srcpkgs/kitty/template:44]: It was from this text in kitty's website:
> You need a couple of extra dependencies to build linux-package. tic to compile terminfo files, usually found in the development package of ncurses.

But yeah, if it compiles properly, then indeed there's no reason to have it there.

Comment (classabbyamp) [reply srcpkgs/kitty/template]: tic compilation should be done at build-time

**Comment by sirkhancision** [comment]
I don't really know why it is failing compilation. I'm building it just fine for my architecture, and even that one fails.

**Comment by paper42** [comment]
> I don't really know why it is failing compilation. I'm building it just fine for my architecture, and even that one fails.

It's failing to run tests, you can run tests by passing -Q to xbps-src. It's also failing to cross compile because it's trying to run the kitty binary which is build for a different architecture than what is the host system. You can reproduce that by passing for example `-a armv7l` to xbps-src.

**Comment by sirkhancision** [comment]
> > I don't really know why it is failing compilation. I'm building it just fine for my architecture, and even that one fails.
> 
> It's failing to run tests, you can run tests by passing -Q to xbps-src. It's also failing to cross compile because it's trying to run the kitty binary which is build for a different architecture than what is the host system. You can reproduce that by passing for example `-a armv7l` to xbps-src.

Sorry, but I'm a bit stumped on how to solve that binary architecture issue. Do you have any ideas on how I could fix that?

**Comment by classabbyamp** [comment]
we'd need to patch in a way to make it use qemu, I think
